crates/blockifier/src/execution/contract_class.rs

```diff
@@ -149,8 +149,9 @@ impl Decode for ContractClassV0Inner {
         input: &mut I,
     ) -> Result<Self, parity_scale_codec::Error> {
         let res = <(Program, Vec<(EntryPointType, Vec<EntryPoint>)>)>::decode(input)?;
+        let entry_point_btree = <BTreeMap<EntryPointType, Vec<EntryPoint>>>::from_iter(res.1);
         let entry_points_by_type =
-            <HashMap<EntryPointType, Vec<EntryPoint>>>::from_iter(res.1.into_iter());
+            <HashMap<EntryPointType, Vec<EntryPoint>>>::from_iter(entry_point_btree);
```

Comment:
Not sure about this. If you converted it to a btree before encoding, decode it as a btree too. Not as as `Vec<(EntryPointType, Vec<EntryPoint>)>`.
And then you just have to iterate once and collect into a `HashMap`